ethcore/src/engines/tendermint/mod.rs

```diff
@@ -590,6 +637,24 @@ impl Engine for Tendermint {
 		None
 	}
 
+	fn epoch_verifier<'a>(&self, _header: &Header, proof: &'a [u8]) -> ConstructedVerifier<'a> {
+		let (signal_number, set_proof) = match destructure_proofs(proof) {
+			Ok(x) => x,
+			Err(e) => return ConstructedVerifier::Err(e),
+		};
+
+		let first = signal_number == 0;
+		match self.validators.epoch_set(first, self, signal_number, set_proof) {
+			Ok((list, _)) => {
+				// Tendermint blocks have instant finality, so no need to check finality proof
+				ConstructedVerifier::Trusted(Box::new(EpochVerifier {
```

Comment:
:+1: will add the logic for checking finality of the block hash returned by `epoch_set`.

Does something need to be done wrt checking validity of the "proof" passed here, or is it sufficient to have that happen at https://github.com/paritytech/parity/blob/master/ethcore/src/snapshot/consensus/authority.rs?utf8=%E2%9C%93#L199?